design/a2a/openapi.yml

```
@@ -0,0 +1,234 @@
+openapi: 3.0.1
+info:
+  title: p4pa-auth
+  description: Api and Models
+  version: 0.0.1
+servers:
+  - url: "http://localhost:8080/payhub"
+paths:
+  /auth/token:
+    post:
+      tags:
+        - authn
+      operationId: postToken
+      security: []
+      parameters:
+        - name: client_id
+          in: query
+          required: true
+          schema:
+            enum: [
+              "piattaforma-unitaria"
+            ]
+            type: string
+        - name: grant_type
+          in: query
+          required: true
+          schema:
+            enum: [
+              "urn:ietf:params:oauth:grant-type:token-exchange",
+              "client_credentials"
+            ]
+            type: string
+        - name: subject_token
+          description: "ID Token"
+          in: query
+          required: true
+          schema:
+            type: string
+        - name: subject_issuer
+          description: "Issuer del subject_token"
+          in: query
+          required: true
+          schema:
+            type: string
+        - name: subject_token_type
+          in: query
+          required: true
+          schema:
+            enum: [
+              "urn:ietf:params:oauth:token-type:jwt"
+            ]
+            type: string
+        - name: scope
+          in: query
+          required: true
+          schema:
+            enum: [
+              "openid"
+            ]
+            type: string
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AccessToken'
+        '400':
+          description: Invalid request
+        '401':
+          description: Authentication failed
+        '429':
+          description: Too Many Requests
+        '500':
+          description: Server ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AuthErrorDTO'
+  /auth/revoke:
+    post:
+      tags:
+        - authn
+      operationId: logout
+      security: []
+      parameters:
+        - name: client_id
+          in: query
+          required: true
+          schema:
+            enum: [
+              "piattaforma-unitaria"
+            ]
+            type: string
+        - name: token
+          description: "Access Token"
+          in: query
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '400':
+          description: Invalid request
+        '401':
+          description: Invalid client_id
+  /am/clients/{organizationIpaCode}:
```

I am not sure that "clients" is a Access Management entity...
I think it should still be an Authentication concept
```suggestion
  /auth/clients/{organizationIpaCode}:
```